src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitDatabase.cs

```diff
@@ -64,7 +64,7 @@ public async Task Execute()
                 _context.NodeDataConsumer = beamSyncProvider.NodeDataConsumer;
             }
 
-            _context.DisposeStack.Push(_context.DbProvider);
+            // _context.DisposeStack.Push(_context.DbProvider);
```

Comment:
what was wrong with it?

Reply:
Db's are already disposed explicitly on shutdown. I moved it explicitly at the end of shutdown. When pushed on dispose stack it could be disposed too early, before other disposible that was relying on DB access (DatabaseMigration).